engine/storage/snapshot/src/main/java/org/apache/cloudstack/storage/snapshot/DefaultSnapshotStrategy.java

```diff
@@ -626,6 +646,30 @@ public StrategyPriority canHandle(Snapshot snapshot, Long zoneId, SnapshotOperat
         return StrategyPriority.DEFAULT;
     }
 
+    private StrategyPriority validateVmSnapshot(Snapshot snapshot) {
+        VolumeVO volumeVO = volumeDao.findById(snapshot.getVolumeId());
+        Long instanceId = volumeVO.getInstanceId();
+        if (instanceId == null) {
+            return StrategyPriority.DEFAULT;
+        }
+
+        VMInstanceVO vm = vmInstanceDao.findById(instanceId);
+        if (vm == null) {
+            return StrategyPriority.DEFAULT;
+        }
+
+        for (VMSnapshotVO vmSnapshotVO : vmSnapshotDao.findByVmAndByType(vm.getId(), VMSnapshot.Type.Disk)) {
+            List<VMSnapshotDetailsVO> vmSnapshotDetails = vmSnapshotDetailsDao.listDetails(vmSnapshotVO.getId());
+            if (vmSnapshotDetails.stream().anyMatch(vmSnapshotDetailsVO -> vmSnapshotDetailsVO.getName().equals(VolumeApiServiceImpl.KVM_FILE_BASED_STORAGE_SNAPSHOT))) {
+                logger.warn("VM [{}] already has KVM File-Based storage VM snapshots. These VM snapshots and volume snapshots are not supported " +
```

Comment:
Hi @JoaoJandre, just a question here - from what I got till now, your feature creates external snapshots, right? Aren't the disk snapshots (take volume snapshot) also external? If both are external, how will this erase the VM snapshots?

Reply:
You are right that both are external. 

However, for each VM snapshot, this feature creates a delta on top of the current volume, these deltas are not copied anywhere, they stay in the VM's backing chain. It is assumed that, when using this feature, if you have VM snapshots, at least one of them is part of the VM's backing chain.

When restoring a volume snapshot, the volume of the VM will be overwritten with the snapshot.

So restoring a volume snapshot would break the previous assumption.

Reply:
Thanks, @JoaoJandre, for the explanation! I've missed that could cause a problem with the synchronisation of the delta here